src/RemoteInboxNotifications/SpecRunner.php

```diff
@@ -0,0 +1,161 @@
+<?php
+/**
+ * Runs a single spec.
+ *
+ * @package WooCommerce Admin/Classes
+ */
+
+namespace Automattic\WooCommerce\Admin\RemoteInboxNotifications;
+
+defined( 'ABSPATH' ) || exit;
+
+use \Automattic\WooCommerce\Admin\Notes\WC_Admin_Note;
+
+/**
+ * Runs a single spec.
+ */
+class SpecRunner {
+	/**
+	 * Run the spec.
+	 *
+	 * @param object $spec The spec to run.
+	 */
+	public static function run_spec( $spec ) {
+		$data_store = \WC_Data_Store::load( 'admin-note' );
+
+		// Create or update the note.
+		$existing_note_ids = $data_store->get_notes_with_name( $spec->slug );
+		if ( 0 === count( $existing_note_ids ) ) {
+			$note = new WC_Admin_Note();
+			$note->set_status( WC_Admin_Note::E_WC_ADMIN_NOTE_PREUNACTIONED );
+		} else {
+			$note = new WC_Admin_Note( $existing_note_ids[0] );
+		}
+
+		// Evaluate the spec and get the new note status.
+		$evaluate_and_get_status = new EvaluateAndGetStatus(
+			new RuleEvaluator(
+				new GetRuleProcessor()
+			)
+		);
+		$status                  = $evaluate_and_get_status->evaluate(
+			$spec,
+			$note->get_status()
+		);
+
+		// Get the matching locale or fall back to en-US.
+		$locale = self::get_locale( $spec->locales );
+
+		// Set up the note.
+		$note->set_title( $locale->title );
+		$note->set_content( $locale->content );
+		$note->set_content_data( (object) array() );
+		$note->set_status( $status );
+		$note->set_type( $spec->type );
+		$note->set_icon( $spec->icon );
+		$note->set_name( $spec->slug );
+		$note->set_source( $spec->source );
+
+		// Clear then create actions.
+		$note->clear_actions();
+		foreach ( $spec->actions as $action ) {
```

Comment:
I also received this error after running the cron event:

```
Undefined property: stdClass::$actions in /app/public/wp-content/plugins/woocommerce-admin/src/RemoteInboxNotifications/SpecRunner.php on line 61
```

Reply:
This is happening because one of the specs is missing an `actions` property. I don't see how, the current JSON file (https://gist.githubusercontent.com/becdetat/f17e9617e5ca1211d0579cf079862905/raw/99495c525c54663f6a0b90edff81e9ccd5b72afb/rinds-specs.json), which hasn't changed for a while, has actions for all of the specs and I can't reproduce this.

Try deleting the `wc_remote_inbox_notifications_specs` option, making sure you're using the JSON from above, and run the cron job again.